pkg/controller/service/service_controller_test.go

```diff
@@ -411,6 +415,38 @@ func TestProcessServiceUpdate(t *testing.T) {
 
 }
 
+// TestConflictWhenProcessServiceUpdate tests if processServiceUpdate will
+// retry creating the load balancer if the update operation returns a conflict
+// error.
+func TestConflictWhenProcessServiceUpdate(t *testing.T) {
+	svcName := "conflict-lb"
+	svc := newService(svcName, types.UID("123"), v1.ServiceTypeLoadBalancer)
+	controller, _, client := newController()
+	client.PrependReactor("update", "services", func(action core.Action) (bool, runtime.Object, error) {
+		update := action.(core.UpdateAction)
+		return true, update.GetObject(), apierrors.NewConflict(action.GetResource().GroupResource(), svcName, errors.New("Object changed"))
+	})
+
+	svcCache := controller.cache.getOrCreate(svcName)
+	if err := controller.processServiceUpdate(svcCache, svc, svcName); err == nil {
+		t.Errorf("controller.processServiceUpdate() = nil, want error")
+	}
+
+	retryMsg := "Error creating load balancer (will retry)"
+	if gotEvent := func() bool {
+		events := controller.eventRecorder.(*record.FakeRecorder).Events
+		for len(events) > 0 {
```

Comment:
Can we do something similar to https://github.com/kubernetes/kubernetes/blob/6ecec23690afaebecb88bd2bebb666ab0665c39e/pkg/cloudprovider/providers/gce/gce_loadbalancer_utils_test.go#L385-L403?

Seems better to have an timeout on this.

Reply:
Everything is synchronous in this test. The time processServiceUpdate is returned, all events should be recorded. And "for len(events) > 0" guards that we will not wait for future (and there won't be future events) here.

Reply:
Makes sense, thanks!